ui-frontend/packages/catalog-ui-search/src/main/webapp/react-component/location/base.line.js

```diff
@@ -188,4 +197,344 @@ const BaseLine = (props) => {
   )
 }
 
+const LineDms = (props) => {
+  const {
+    geometryKey,
+    dmsPointArray,
+    setState,
+    unitKey,
+    setBufferState,
+    widthKey,
+  } = props
+  const [baseLineError, setBaseLineError] = useState(initialErrorState)
+  const [bufferError, setBufferError] = useState(initialErrorState)
+
+  useEffect(() => {
+    if (props.drawing) {
+      setBaseLineError(initialErrorState)
+    }
+    if (dmsPointArray) {
+      setBaseLineError(validateDmsLineOrPoly(dmsPointArray, geometryKey))
+    }
+  }, [props.polygon, props.line, dmsPointArray])
+
+  return (
+    <div>
+      <div className="input-location">
+        {dmsPointArray &&
+          dmsPointArray.map((point, index) => {
+            return (
+              <div>
+                <DmsTextField
+                  key={'point-' + index}
+                  point={point}
+                  setPoint={(point) => {
+                    let array = [...dmsPointArray]
+                    array.splice(index, 1, point)
+                    setState({ ['dmsPointArray']: array })
```

Comment:
📝  Copying the behavior of the other geo inputs, I am setting the value as soon as a user types it whether or not its valid and then this component will validate it as the state changes. 
The reason we want to do this is so that a user can freely type without snapping and see the map get automatically updated with their points as they become valid.

Reply:
📝  Another note about setting the state here, this is an interesting section of code because the "state" here is a backbone model (location-old). So when this state is set, it triggers an 'onChange' on the model and when a change happens on the model, it updates the state.